library/src/main/java/com/github/gfx/android/orma/OrmaConnection.java

```diff
@@ -93,17 +94,21 @@ public OrmaConnection(@NonNull OrmaDatabaseBuilderBase<?> builder, List<Schema<?
         this.trace = builder.trace;
         this.readOnMainThread = builder.readOnMainThread;
         this.writeOnMainThread = builder.writeOnMainThread;
-        this.db = openDatabase(builder.context);
+        this.db = openDatabase(builder);
 
         checkSchemas(schemas);
     }
 
-    private SQLiteDatabase openDatabase(Context context) {
-        SQLiteDatabase db;
+    private Database openDatabase(OrmaDatabaseBuilderBase<?> builder) {
+        Database db;
         if (name == null) {
-            db = SQLiteDatabase.create(null);
+            db = new DefaultDatabase(SQLiteDatabase.create(null));
         } else {
-            db = context.openOrCreateDatabase(name, openFlags(), null, null);
+            DatabaseProvider provider = builder.databaseProvider;
+            if (provider == null) {
+                provider = new DefaultDatabase.Provider();
```

Comment:
Please set the default provider in OrmaDatabaseBuilderBase; so OrmaConnection assumes provider is not null.